### PR TITLE
Allow multiple instances of ccx

### LINKED ISSUE #175: Bug: when creating a CCX via REST API, the coach can't access the coach dashboard
First guess: the coach role isn't being created properly.

### PR DESCRIPTION
**Pull request by amir-qayyum-khan**
### Background
This work fixes #175 
Fixes bug where coach with multiple CCXs can't view coach dashboard.

**LMS Updates:** 
- Now a coach can have multiple instance of ccx on same master course.

@pdpinch @giocalitri @justinabrahms  @pwilkins

### CODE REVIEW

**Comment by pdpinch** [comment]
@amir-qayyum-khan how can we best test this?

**Comment by amir-qayyum-khan** [comment]
@pdpinch you can test it via https://github.com/mitocw/edx-platform/pull/181

**Comment by amir-qayyum-khan** [comment]
I think this PR is useless. We should close its a use #181 instead. 
#181 has form to create ccx and it also allow to use multiple instances of ccx on same course by same coach.

**Comment by giocalitri** [comment]
I do not think it is useless and it can be easily tested using the REST APIs

File: lms/djangoapps/ccx/utils.py [hunk 1]
```diff
@@ -128,6 +128,30 @@ def get_ccx_for_coach(course, coach):
...
+    """
+    Looks to see if user is coach of a CCX for this course.  Returns the CCX or
+    None.
+
+    Arguments:
+        course (CourseDescriptor): Master course
+        coach (User): Coach to ccx
+        ccx_id (long): Id of ccx
+
+    Returns: Instance of CCX.
```
Comment (giocalitri) [inline lms/djangoapps/ccx/utils.py]: nit: please put `Instance of CCX` on a new line and specify the object type

**Comment by giocalitri** [comment]
I think this PR looks fine overall, but I would like to know what happens with [this line of code](https://github.com/mitocw/edx-platform/blob/feature/aq/support_mutiple_instances_ccx_mitocw%23175/lms/djangoapps/ccx/views.py#L117) when there are more than one CCX

**Comment by amir-qayyum-khan** [comment]
Done @giocalitri

**Comment by giocalitri** [comment]
@amir-qayyum-khan I do not understand why you changed that line. Does it mean that now every CCX is pulled by ID? So what's ne need of the `get_ccx_for_coach`? Is it used anywhere?

**Comment by amir-qayyum-khan** [comment]
@giocalitri we get_ccx_for_coach dont  need this, 
Because all view come from ```coach_dashboard```. This only means that if controller dont have ccx then redirect user to his first ccx on the given course.

**Comment by giocalitri** [comment]
@amir-qayyum-khan so, is `get_ccx_for_coach` still used or not in the codebase?

**Comment by amir-qayyum-khan** [comment]
@giocalitri this will be used in current scenario, (But It will be fixed in #181)
Because for a master course when coach clicks on ``ccx coach`` tab. 
I current context we take him to his first ccx, But this will be fixed in #181.

cc @pdpinch

**Comment by giocalitri** [comment]
@amir-qayyum-khan where is it used?

**Comment by amir-qayyum-khan** [comment]
in coach dashboard view,  the view which renders coach dashboard or create
form base on ccx availability

**Comment by giocalitri** [comment]
@amir-qayyum-khan can you please pull the current master and rebase this on the current master?

**Comment by giocalitri** [comment]
when you are done I'll check the functionality and, if everything works, we can open against edx

**Comment by amir-qayyum-khan** [comment]
Done @giocalitri

**Comment by giocalitri** [comment]
This is good to go upstream :+1:

**Comment by pdpinch** [comment]
I keep confusing this with #181, so I changed the title. It's clearer for me. I hope I haven't made it less clear for everyone else.

**Comment by amir-qayyum-khan** [comment]
it seem ok to me. :+1:

**Comment by amir-qayyum-khan** [comment]
Made PR on upstream https://github.com/edx/edx-platform/pull/11545

**Pull request closed without merging**